oauthproxy.go

```diff
@@ -961,15 +961,57 @@ func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.R
 //
 //nolint:gosimple
 func authOnlyAuthorize(req *http.Request, s *sessionsapi.SessionState) bool {
-	// Allow secondary group restrictions based on the `allowed_groups`
-	// querystring parameter
-	if !checkAllowedGroups(req, s) {
-		return false
+	constraints := []func(*http.Request, *sessionsapi.SessionState) bool{
+		checkAllowedGroups,
+		checkAllowedEmailDomains,
+	}
+
+	for _, constraint := range constraints {
+		if !constraint(req, s) {
+			return false
+		}
 	}
 
 	return true
 }
 
+// checkAllowedEmailDomains allow email domain restrictions based on the `allowed_email_domains`
+// querystring parameter
+func checkAllowedEmailDomains(req *http.Request, s *sessionsapi.SessionState) bool {
+	allowedEmailDomains := extractAllowedEmailDomains(req)
+	if len(allowedEmailDomains) == 0 {
+		return true
+	}
+
+	emailSplited := strings.Split(s.Email, "@")
+	if len(emailSplited) != 2 {
+		return false
+	}
+
+	if _, ok := allowedEmailDomains[emailSplited[1]]; ok {
```

Comment:
Throughout the rest of the codebase, there's support for board subdomains (`.example.com` would allow `foo.example.com` and `bar.example.com`).  I think that support should be extended here too.

Reply:
I believe there's common helper methods for this logic since it is an error-prone implementation that leads to security vulnerabilities; hopefully we can use those here?

Reply:
It support now broad subdomains starting with `.` and `*.` and sharing the same code base through an util.